lib/demo_web/live/pong/pong.ex

```diff
@@ -0,0 +1,212 @@
+      <div class="game-over" style="position: absolute; margin:auto; top: 200px; width:<%= @width %>px; z-index: 20; color: white">
+        <h1>GAME OVER</h1>
+        <h2>SCORE: <%= @ball.score %></h2>
+        <button phx-click="new_game">NEW GAME</button>
+      </div>
+    <% end %>
+    <svg
+      class="pong-container"
+      phx-keydown="keydown"
+      phx-keyup="keyup"
+      phx-target="window"
+      transform="scale(1,-1)"
+      height="<%= @height %>"
+      width="<%= @width %>"
+      style="
+        background-color: #222;
+      "
+    >
+      <circle
+        class="ball"
+        cx="<%= @ball.position.x %>"
+        cy="<%= @ball.position.y %>"
+        r="<%= @ball.size/2 %>"
+        fill="white"
+      />
+
+      <rect
+        class="paddle player-1"
+        x="<%= @player_1_position - @paddle_size/2 %>"
+        y="<%= @paddle_offset - 20 %>"
+        width="<%= @paddle_size %>"
+        height="20"
+        fill="white"
+      />
+      <rect
+        class="paddle player-2"
+        x="<%= @player_1_position - @paddle_size/2 %>"
+        y="<%= @height - @paddle_offset %>"
+        width="<%= @paddle_size %>"
+        height="20"
+        fill="white"
+      />
+    </svg>
+    """
+  end
+
+  def mount(_session, socket) do
+    {:ok, socket |> new_game() |> schedule_tick()}
+  end
+
+  defp new_game(socket) do
+    defaults = %{
+      score: 0,
+      game_state: :playing,
+      heading: :stationary,
+      pending_headings: %{left: false, right: false},
+      ball: %Ball{
+        size: @ball_size,
+        bounds: %{
+          x: %{min: 0, max: @width},
+          y: %{min: @paddle_offset, max: @height - @paddle_offset},
+        },
+      },
+      paddle_size: @paddle_size,
+      paddle_offset: @paddle_offset,
+      width: @width,
+      height: @height,
+      tick: @tick,
+      player_1_position: @player_1_position,
+    }
+
+    socket
+    |> assign(defaults)
+  end
+
+  def handle_event("update_settings", %{"width" => width, "tick" => tick}, socket) do
+    {width, ""} = Integer.parse(width)
+    {tick, ""} = Integer.parse(tick)
+
+    new_socket =
+      socket
+      |> update_size(width)
+      |> update_tick(tick)
+
+    {:noreply, new_socket}
+  end
+
+  def handle_event("new_game", _, socket) do
+    {:noreply, new_game(socket)}
+  end
+
+  def handle_event("keydown", key, socket) do
+    direction = case key do
+      "ArrowLeft" -> :left 
+      "ArrowRight" -> :right
+      _ -> :none
+    end
+    {:noreply, set_heading(socket, direction)}
+  end
+
+  def handle_event("keyup", key, socket) do
+    direction = case key do
+      "ArrowLeft" -> :left 
+      "ArrowRight" -> :right
+      _ -> :none
+    end
+    {:noreply, unset_heading(socket, direction)}
+  end
+
+  def handle_info(:tick, socket) do
+    new_socket =
+      socket
+      |> game_loop()
+      |> schedule_tick()
+
+    {:noreply, new_socket}
+  end
+
+  defp update_tick(socket, tick) when tick <= 1000 and tick >= 50 do
```

Comment:
Is this useless? your tick is always 20. This is never called

Reply:
This was a remnant of the drop-down which controlled tick rate, removed the box but not this :shrug: